homeassistant/components/airtouch5/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow for Airtouch 5 integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from airtouch5py.airtouch5_simple_client import Airtouch5SimpleClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required("host"): str})
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    client = Airtouch5SimpleClient(data["host"])
+
+    try:
+        await client.test_connection()
+    except Exception:
+        raise CannotConnect
+
+    # Return info that you want to store in the config entry.
+    return {"title": data["host"]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Airtouch 5."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] | None = None
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors = {"base": "cannot_connect"}
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors = {"base": "unknown"}
+            else:
+                return self.async_create_entry(title=info["title"], data=user_input)
```

Comment:
```suggestion
                await self.async_set_unique_id(user_input[CONF_HOST])
                self._abort_if_unique_id_configured()
                return self.async_create_entry(title=user_input[CONF_HOST], data=user_input)
```

By setting the unique id and aborting, we allow only one device per host